src/resources/js/parts-kit.js

```diff
@@ -174,13 +114,148 @@ class Toggle {
         typeof this.config[prop] === "function" && this.config[prop]();
     }
 
-    handleKeydown = ({ keyCode }) => {
-        if (this.isOpen && keyCode === keyCodes.esc) {
+    handleKeydown = ({ key }) => {
+        if (this.isOpen && key === "Escape") {
             this.close();
         }
     };
 }
 
-// Init the toggle class
-const toggles = document.querySelectorAll("[data-parts-kit-toggle]");
-toggles.forEach((el) => new Toggle(el));
+// Parts kit class for expanding/collapsing sidebar
+class PartsKit {
+    constructor(el) {
+        this.el = el;
+        this.setVars();
+        this.bindEvents();
+        this.initSidebar();
+    }
+
+    setVars() {
+        this.sidebar = this.el.querySelector("[data-sidebar]");
+        this.lsSidebar = "pk-sidebar-hidden";
+        this.fullscreenExpandBtn = this.el.querySelector(
+            "[data-fullscreen-expand]"
+        );
+        this.fullscreenCloseBtn = this.el.querySelector(
+            "[data-fullscreen-close]"
+        );
+
+        // Set up toggle classes
+        const toggleElements = document.querySelectorAll("[data-parts-kit-toggle]");
+        this.toggles = [...toggleElements].map(el => new Toggle(el));
+
+        /** @type {HTMLInputElement} */
```

Comment:
ah, how have you been liking jsdoc typing?

Reply:
I did it quite a bit at my old job. I found it to be helpful. I really like using it for HTML elements because I don't have to check MDN as often for tiny things I can't remember.